### PR TITLE
Bool to return reward for BruteForce and Optimise sensor managers

### PR DESCRIPTION
**Pull request by asaunders-dstl**
Added return_reward, declared when performing chose_actions. Returns the chosen_actions, reward when True

### CODE REVIEW

File: stonesoup/sensormanager/base.py [hunk 1]
```diff
@@ -111,7 +111,8 @@ def choose_actions(self, tracks, timestamp, nchoose=1, **kwargs):
             Time at which the actions are carried out until
         nchoose : int
             Number of actions from the set to choose (default is 1)
-
+        return_reward: bool
+            Whether to return the reward for chosen actions (default is False)
         Returns
         -------
         : dict
```
Comment (nperree-dstl) [inline stonesoup/sensormanager/base.py:118]: Could you add something here to describe how the output changes if return_reward=True? I.e. it returns a list of dictionaries and the reward (as a numpy array usually of 1D?).

File: stonesoup/sensormanager/base.py [hunk 2]
```diff
@@ -111,7 +111,9 @@ def choose_actions(self, tracks, timestamp, nchoose=1, **kwargs):
             Time at which the actions are carried out until
         nchoose : int
             Number of actions from the set to choose (default is 1)
-
+        return_reward: bool
+            Whether to return the reward for chosen actions (default is False)
+            When True, returns a tuple of 1d arrays: (dictionaries of chosen actions, rewards)
         Returns
         -------
         : dict
```
Comment (nperree-dstl) [inline stonesoup/sensormanager/base.py:119]: Sorry I'm being picky...

```suggestion
        return_reward: bool
            Whether to return the reward for chosen actions (default is False).
            When True, returns a tuple of 1d arrays: (dictionaries of chosen actions, rewards)
        Returns
        -------
        : list(dict) or (list(dict), :class:`numpy.ndarray`)
```

Then probably also clarify in the line below that the dict is the configuration of ":class:`~.Sensor`:` [:class:`~.Action`]"and the array contains the corresponding reward.

File: stonesoup/sensormanager/base.py [hunk 3]
```diff
@@ -111,11 +111,13 @@ def choose_actions(self, tracks, timestamp, nchoose=1, **kwargs):
...
+        return_reward: bool
+            Whether to return the reward for chosen actions (default is False)
+            When True, returns a tuple of 1d arrays: (dictionaries of chosen actions, rewards)
         Returns
         -------
-        : dict
-            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected
+        : list(dict) or (list(dict), :class:`numpy.ndarray`)
+            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected and the array contains the corresponding reward.
         """
```
Comment (nperree-dstl) [inline stonesoup/sensormanager/base.py:122]:
```suggestion
            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected and the array contains 
            the corresponding reward.
        """

```

Comment (nperree-dstl) [reply stonesoup/sensormanager/base.py:122]: Too many characters on one line

File: stonesoup/sensormanager/base.py [hunk 4]
```diff
@@ -111,11 +111,14 @@ def choose_actions(self, tracks, timestamp, nchoose=1, **kwargs):
...
-
+        return_reward: bool
+            Whether to return the reward for chosen actions (default is False)
+            When True, returns a tuple of 1d arrays: (dictionaries of chosen actions, rewards)
         Returns
         -------
-        : dict
-            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected
+        : list(dict) or (list(dict), :class:`numpy.ndarray`)
+            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected and the array contains
```
Comment (nperree-dstl) [inline stonesoup/sensormanager/base.py:120]:
```suggestion
            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected and the array contains
```

**Review by nperree-dstl — APPROVED** [review]

**Review by jswright-dstl — APPROVED** [review]

**Pull request merged**